src/java_time/data.clj

```diff
@@ -0,0 +1,124 @@
+(ns java-time.data
+  "A pure-data view of all the important/data-holding java.time objects."
+  (:require [clojure.core.protocols :as p])
+  (:import (java.time YearMonth Month DayOfWeek Instant
+                      LocalTime LocalDateTime
+                      ZonedDateTime OffsetDateTime)))
+
+(extend-protocol p/Datafiable
```

Comment:
My initial thought would be to use the `java-time.properties` machinery to create the data representation of the date-time entities. That way you wouldn't need to come up with property names and cherry-pick values by hand. Did you consider the "properties" approach?

Reply:
Ok, ignoring the current (reflective) operations in properties.clj, here are my thoughts:

I personally prefer the direct nature of the current solution. Moreover, if I was to return all the possible iso,chrono,julian etc fields, that would make the whole thing much heavier, as a lot of these require some sort of calculation. Cherry-picking (as you put it) the public members of each object separately means that, for the most part, we get back static values (super quickly). 

As for naming, I guess a mechanized approach of lower-case + dashize would work, but again, that's essentially what I've done (albeit manually), and I do like how straight-forward it reads. I guess I can rework that if you insist, but I would avoid tying this namespace with individual Field access (in properties.clj). 

Thanks for you time reviewing :)

Reply:
Even if you create the names for the properties manually, I'd stick with a mechanistic approach of taking the Java name and "clojureizing" it. This way the user can have a good
chance of guessing what's inside the map. This approach also retains consistency with `properties.clj`.

Reply:
Ok, I'll to do that, but I can immediately see some complications...For example what do I do with `month-name` or `day-name`, or `leap-year?`? Basically, for some of my keys there are no direct Fields to draw names from. How do you want to handle those? Would you like a mixed approach, where some keys would be hard-coded, and some would be mechanically transformed from the corresponding Fields? I find this unnecessarily complex, but if that's what you prefer that's fine...just let me know :+1:

Reply:
Maybe we're going about it completely the wrong way...what if instead of a flat map we returned a nested map like this:
```clj
{:era {:year ...} ;; year-of-era 
 :year {:value 2020 
           :leap? true
           :quarter 1;; quarter-of-year
           :week 1 ;; week-of-year
           :month 1 ;; month-of-year 
           :day 8}    ;; day-of-year
 :month {:name "January"
              :length 31}
 :week {:day 2} ;; day-of-week
 :day {:name "Monday"}
...
...
}
```
An approach like this would justify the names, in the sense that a ChronoField like `YEAR_OF_ERA` would be found under path `[:era :year]`, `DAY_OF_YEAR` under `[:year :day]` etc etc. This is rather nice, and idiomatic I might add. It diverges from `properties.clj` even further, but it kind of feels like the right thing to do.

Let me know of your thoughts...

Reply:
Here is what it looks like:

```clj
(datafy 
  (OffsetDateTime/now))
=>
{:year {:month {:name "JANUARY", 
                :value 1, 
                :length 31
                :day 6}, 
        :length 366, 
        :leap? true, 
        :value 2020, 
        :day 6},
 :week   {:day {:name "MONDAY", :value 1}},
 :day    {:hour 18},
 :hour   {:minute 26},
 :minute {:second 48},
 :second {:nano 963253000, :milli 963, :micro 963253},
 :local-time "18:26:48.963253",
 :offset {:id "Z", :hours 0, :seconds 0}}
```

Hope that helps ;)

Reply:
I like this. Better than the previous version.